singer/messages.py

```diff
@@ -37,9 +37,10 @@ class RecordMessage(Message):
 
     '''
 
-    def __init__(self, stream, record, version=None):
+    def __init__(self, stream, record, time_extracted=None, version=None):
```

Comment:
You should add time_extracted after version in case someone is calling `RecordMessage(stream, record, version)`.

Reply:
We should require time_extracted to be an "aware" datetime.datetime object. Maybe just add a check here to validate that it's either None or a datetime.datetime with a tzinfo property.